Python/ast.c

```diff
@@ -925,6 +925,17 @@ ast_for_funcdef(struct compiling *c, const node *n, asdl_seq *decorator_seq)
     body = ast_for_suite(c, CHILD(n, name_i + 3));
     if (!body)
         return NULL;
+    
+    n = asdl_seq_LEN(body);
```

Comment:
I'm not sure I understand how this works. Will it warn for *every* list comprehension or only those where a variable defined in the list comprehension also is a free variable in the parent function? Maybe you just need to explain to me what `asdl_seq_GET` does?

Reply:
This is what is happening:
- We inspect a function definition, `ast_for_funcdef`
- If we encounter a list comprehension, `islistcomp = compiler_islistcomp(st);`
- Then warn that a variable attached to list comp will leak in 3.x

This: 

`asdl_seq_GET`  

gets a statement, in this case from the function body `s->v.FunctionDef.body`. Then we check if it is a `listcomp` on the next statement.

Reply:
Will this warn for e.g.:

```python
def f():
  x = 0
  print([y for y in [0,1,2]])
  print(x)
```

?

Reply:
Yes

Reply:
That would be a false positive?